woocommerce/woocommerce
# Template API: Use expression evaluation in woocommerce/conditional

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR updates the `woocommerce/conditional` block, and its usage, to support the `showIf` attribute.

Specifically, the acceptance criteria from #40678 are implemented:

- woocommerce/conditional block is updated to support a `showIf` attribute that contains an expression, that if evaluated to `true`, shows the inner blocks of the conditional block
- woocommerce/conditional block is updated to deprecate the `mustMatch` attribute
- All uses of woocommerce/conditional in the simple product template are updated to use `showIf`
- All uses of woocommerce/conditional in the variation product template are updated to use `showIf`

Closes #40678.



### How to test the changes in this Pull Request:



1. Enable the **Product Block Editor** under **WooCommerce** > **Settings** > **Advanced** > **Features** > **Experimental features**.
2. Go to **Products** > **Add New**
3. Check if the `@woocommerce/expression-evaluation` package is loaded via dependency injection by confirming that the `wc.expressionEvaluation` global is available when the product editor page is loaded. In the browser development console:

```js
wc.expressionEvaluation.evaluate( '1 + 3' )
```

4. Check that the uses of the `woocommerce/conditional` block are correct (no functional regression/change). The "Inventory" tab makes use of this block:

Track stock quantity disabled:

<img width="1255" alt="Screenshot 2023-10-10 at 15 10 19" src="https://github.com/woocommerce/woocommerce/assets/2098816/69bfb276-53c3-486f-ab83-be200bc61425">

Track stock quantity enabled:
<img width="1255" alt="Screenshot 2023-10-10 at 15 10 39" src="https://github.com/woocommerce/woocommerce/assets/2098816/26818e00-88bf-490a-92ef-3b6022d327b3">



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
